libgnucash/engine/gnc-datetime.cpp

```diff
@@ -353,17 +384,23 @@ tz_from_string(std::string str)
     return TZ_Ptr(new PTZ(tzstr));
 }
 
-GncDateTimeImpl::GncDateTimeImpl(std::string str) :
+GncDateTimeImpl::GncDateTimeImpl(const char* str) :
     m_time(unix_epoch, utc_zone)
 {
-    if (str.empty()) return;
+    if (!str || !str[0]) return;
     TZ_Ptr tzptr;
     try
     {
+        if (auto res = fast_iso8601_utc_parse (str))
+        {
+            static const TZ_Ptr utc_tz = tz_from_string("+0000");
```

Comment:
Use [`utz_zone`](https://github.com/Gnucash/gnucash/blob/57e3b6f57661964f036192e76e44c707a50de645/libgnucash/engine/gnc-datetime.cpp#L67) instead of creating a new one.